test/parallel/test-repl-clear-immediate-crash.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+const common = require('../common');
+const child_process = require('child_process');
+const assert = require('assert');
+
+// Regression test for https://github.com/nodejs/node/issues/37806:
+const proc = child_process.spawn(process.execPath, ['-i']);
+proc.on('error', common.mustNotCall());
+proc.on('exit', common.mustCall((code) => {
+  assert.strictEqual(code, 0);
+}));
+proc.stdin.write('clearImmediate({});\n.exit\n');
```

Comment:
This test is not really required, since it was just caused by the crash but it does not hurt either.

Reply:
The issue, I think, was a bit different (if by "the crash" you mean that NRE for this._idleNext). The issue was caused because an `undefined` `async_id` was given to `emitDestroy`, and this caused an error in `emitDestroyScript` (as the check there only checks `<= 0` and undefined is not `<= 0`).

Reply:
Maybe a better fix would've been to actually change the code there to check (in `async_hooks.js`) to `hasHooks(kDestroy) && asyncId > 0` instead of an early return - but that's probably out of scope for this minor issue, and maybe a crash there is better than "swallowing" bad async_ids which might cover errors elsewhere.